packages/x-charts/src/Gauge/GaugeContainer.tsx

```diff
@@ -25,9 +25,9 @@ export interface GaugeContainerProps
 const ResizableContainer = styled('div', {
   name: 'MuiGauge',
   slot: 'Container',
-})<{ ownerState: Pick<GaugeContainerProps, 'width' | 'height'> }>(({ ownerState, theme }) => ({
-  width: ownerState.width ?? '100%',
-  height: ownerState.height ?? '100%',
+})<{ ownerState: Pick<GaugeContainerProps, 'width' | 'height'> }>(({ theme }) => ({
+  width: 'var(--width, 100%)',
+  height: 'var(--height, 100%)',
```

Comment:
This seems fragile though:

```jsx
<div style={{ '--width': 500 }}>
  <div
    style={{
      // '--width': 200,
      
      // If forget to set the variable above, the width below won't fall back to `100%`
      // since CSS variables are inherited and the value of 500 will be used
      
      width: 'var(--width, 100%)',
    }}>
  </div>
</div>
```

Does this imply that CSS Variable names should be unique?
cc @siriwatknp @DiegoAndai

Reply:
We haven't settled on a pattern in Material UI yet (will probably do for v7), but I would say yes, a unique name would be useful to avoid future bugs. In most instances, we're prefixing the component name to the variable, for example: https://github.com/mui/material-ui/blob/next/packages/mui-material/src/AppBar/AppBar.js#L100